lib/devicetrust/native/device_darwin.go

```diff
@@ -91,21 +100,107 @@ func pubKeyToCredential(id string, pubKeyRaw []byte) (*devicepb.DeviceCredential
 
 func collectDeviceData() (*devicepb.DeviceCollectedData, error) {
 	var dd C.DeviceData
-	defer func() { C.free(unsafe.Pointer(dd.serial_number)) }()
+	defer func() {
+		C.free(unsafe.Pointer(dd.serial_number))
+		C.free(unsafe.Pointer(dd.model))
+		C.free(unsafe.Pointer(dd.os_version_string))
+	}()
 
 	if res := C.DeviceCollectData(&dd); res != 0 {
 		return nil, trace.Wrap(statusErrorFromC(res))
 	}
 
+	osUser, err := user.Current()
+	if err != nil {
+		return nil, trace.Wrap(err, "reading current user")
+	}
+
+	// Run exec-ed commands concurrently.
+	var wg sync.WaitGroup
+	// Note: We could read the OS build from dd.os_version_string, but this
+	// requires no string parsing.
+	var osBuild, jamfVersion, macosEnrollmentProfiles string
+	for _, spec := range []struct {
+		fn   func() (string, error)
+		out  *string
+		desc string
+	}{
+		{fn: getOSBuild, out: &osBuild, desc: "macOS build"},
+		{fn: getJamfBinaryVersion, out: &jamfVersion, desc: "Jamf version"},
+		{fn: getMacosEnrollmentProfiles, out: &macosEnrollmentProfiles, desc: "macOs enrollment profiles"},
+	} {
+		spec := spec
+		wg.Add(1)
+		go func() {
```

Comment:
Nice inclusion of concurrency for this collection - I probably ought to introduce this for device data collection on Windows 🐎

Reply:
Thanks. I wasn't sure whether it was a great idea or overkill, but since I had the loop it seemed easy enough to do.